src/router/router.js

```diff
@@ -108,8 +111,20 @@ export const navigate = async function () {
     const previousRoute = currentRoute
     const hash = getHash()
     const route = matchHash(hash, this.parent[symbols.routes])
-
+    let beforeHookOutput
     if (route) {
+      if (route.hooks) {
+        if (route.hooks.before) {
+          const args = route.params ? route.params : {}
+          beforeHookOutput = await route.hooks.before(args)
+          if (isString(beforeHookOutput)) {
+            to(beforeHookOutput)
```

Comment:
We should probably pass in the data and options as well in the redirect

Reply:
In that scenario, we can support two variants of redirections. 

- The first one is if beforeHookOutput is of type string, `return "/abc"` then ignoring data and option arguments in the router.to() function. 

- The second one is if beforeHookOutput is an object with the 'path' key being mandatory to represent the path to the target/redirect route, along with 'data' and 'options' keys for respective data.  `return {path: "/abc", data: {a: 1, b:2}, optiosn: {inHistory: true}`
      -  But a major problem with the second variant comes in differentiating between API data/object keys that are expected as props in the component. Here, we should strictly follow the rule that the existence of the 'path' key in the object is treated as redirection, otherwise, it is considered as an object for injecting into component props